File: 프로그래머스/1/138477. 명예의 전당 （1）/명예의 전당 （1）.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

vector<int> solution(int k, vector<int> score) {
    vector<int> r;
    vector<int> t;
    for(int i=0;i<score.size();i++) {
        if(i<k)
            t.push_back(score[i]);
        else {
            for(int j=0;j<t.size();j++) {
                if(score[i]>t[j]) {
                    t[j]=score[i];
                    break;
                }
            }
        }
        sort(t.begin(),t.end(),less<int>());
        r.push_back(t[0]);
    }
    return r;
}

File: 프로그래머스/1/138477. 명예의 전당 （1）/명예의 전당 （1）.py
def solution(k, score):
    r,t=[],[]
    for i,s in enumerate(score):
        if i<k:
            t+=[s]
        else:
            for j,n in enumerate(t):
                if s>n:
                    t[j]=s
                    break
        t.sort()
        r+=[t[0]]
    return r